Bio/SeqIO/PdbIO.py

```diff
@@ -17,6 +17,17 @@
 from .Interfaces import SequenceIterator
 
 
+def _res2aacode(residue, undef_code="X"):
+    """Return a residue's type as a one-letter code.
+
+    Non-standard residues (e.g. CSD, ANP) are returned as 'X'.
+    """
+    onecode = protein_letters_3to1
```

Comment:
You can simply make this `onecode = protein_letters_3to1_extended`, saves you the trouble of the update and retains exactly the same result as you'd have with `seq1`

Reply:
But, `protein_letters_3to1_extended` contains a lot lesser mappings than `protein_letters_3to1`. Will it be okay to do so?

Reply:
I've tried using `onecode = protein_letters_3to1_extended`, but it is causing tests to fail.

Reply:
That's interesting because it's what `seq1` uses internally. Can you double check that the errors come from this change and not from something else?

Reply:
Yes, I've checked. Initially, the `custom_map` argument was set to `protein_letters_3to1` in `restype` function while calling `seq1`. This combined both the dictionaries, and this combined dictionary was then used.

Reply:
Ah... I know what's up. This is confusing. We actually have two mappings called `protein_letters_3to1`, one in `Bio.Data.IUPACData` and one in `Bio.Data.SCOPData`. And one uses capitalized residue names while the other one does not.. This is messy. Let's keep it like this then, makes total sense.

Reply:
Actually, let's do this: we can import both and merge them at the top of the file:

```python
from Bio.Data.SCOPData import protein_letters_3to1 as scop_3to1
from Bio.Data.IUPACData import protein_letters_3to1_extended as iupac_3to1_ext

...

_aa3to1_dict = {}
_aa3to1_dict.update(iupac_3to1_ext)
_aa3to1_dict.update(scop_3to1)


def _res2aacode(...):  # uses _aa3to1_dict
    ...
```

Reply:
This way we get both dictionaries in, just like `seq1` and we keep them in sync with those dicts in other modules (no manual updates here).